pipeline.wdl

```diff
@@ -0,0 +1,246 @@
+version 1.0
+
+# Copyright (c) 2019 Sequencing Analysis Support Core - Leiden University Medical Center
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import "sample.wdl" as sampleWorkflow
+import "structs.wdl" as structs
+import "tasks/common.wdl" as common
+import "tasks/talon.wdl" as talon
+import "tasks/transcriptclean.wdl" as transcriptClean
+
+workflow Pipeline {
+    input {
+        File sampleConfigFile
+        String outputDirectory = "."
+        File annotationGTF
+        String genomeBuild
+        String annotationVersion
+        File referenceGenome
+        String sequencingPlatform
+        File talonConfigFile
+        String pipelineRunName
+        File dockerImagesFile
+
+        File? talonDatabase
+        File? spliceJunctionsFile
+        String? novelIDprefix
+        Int? minimumLength
+        Int? cutoff5p
+        Int? cutoff3p
+        Boolean? filterTranscriptsAbundance = false
+        File? filterPairingsFileAbundance
+        File? summaryDatasetGroupsCSV
+        Int? minIntronSize
+    }
+
+    call common.YamlToJson as convertDockerImagesFile {
+        input:
+            yaml = dockerImagesFile,
+            outputJson = outputDirectory + "/dockerImages.json"
+    }
+
+    Map[String, String] dockerImages = read_json(convertDockerImagesFile.json)
+
+    call SampleConfigToSampleReadgroupLists as convertSampleConfig {
+        input:
+            yaml = sampleConfigFile,
+            outputJson = outputDirectory + "/samples.json",
+            dockerImage = dockerImages["pyyaml"]
+    }
+
+    SampleConfig sampleConfig = read_json(convertSampleConfig.json)
+    Array[Sample] allSamples = sampleConfig.samples
+
+    Boolean runInitDatabase = defined(talonDatabase)
+    Boolean runGetSJsFromGTF = defined(spliceJunctionsFile)
+
+    if (runInitDatabase == false) {
```

Comment:
```suggestion
    if ( ! runInitDatabase) {
```

The name of this variable is kind of misleading, it sounds like the databe initialization will be run if this is true, but right now it will be run if this is false.

Reply:
Renamed to "userProvidedDatabase".